src/rust/engine/hashing/src/lib.rs

```diff
@@ -128,6 +128,18 @@ impl Serialize for Digest {
   }
 }
 
+impl From<&[u8]> for Digest {
```

Comment:
This is a kind of weird use of `From`; `From` is intended for conversions, rather than processing... My assumption from the type here here was that this was taking a serialized `Digest` and deserializing it, rather than digesting the contents of the slice... Maybe this could be `Digest.of_bytes(&[u8])` rather than a `From` impl?

Reply:
+1